app/lib/features/tasks/pages/tasks_list_page.dart

```diff
@@ -54,42 +58,46 @@ class _TasksListPageConsumerState extends ConsumerState<TasksListPage> {
     final spaceId = widget.spaceId;
     return AppBar(
       centerTitle: false,
-      title: Column(
-        mainAxisAlignment: MainAxisAlignment.start,
-        crossAxisAlignment: CrossAxisAlignment.start,
-        children: [
-          Text(lang.tasks),
-          if (spaceId != null) SpaceNameWidget(spaceId: spaceId),
-        ],
-      ),
+      title: widget.onSelectTaskListItem != null
+          ? Text(L10n.of(context).selectTaskList)
+          : Column(
+              mainAxisAlignment: MainAxisAlignment.start,
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Text(lang.tasks),
+                if (spaceId != null) SpaceNameWidget(spaceId: spaceId),
+              ],
+            ),
       actions: [
-        ValueListenableBuilder(
-          valueListenable: showCompletedTask,
-          builder: (context, value, child) {
-            return TextButton.icon(
-              onPressed: () => showCompletedTask.value = !value,
-              icon: Icon(
-                value
-                    ? Icons.visibility_off_outlined
-                    : Icons.visibility_outlined,
-                size: 18,
-              ),
-              label: Text(value ? lang.hideCompleted : lang.showCompleted),
-              style: OutlinedButton.styleFrom(
-                padding: const EdgeInsets.symmetric(horizontal: 10),
-              ),
-            );
-          },
-        ),
-        AddButtonWithCanPermission(
-          key: TasksListPage.createNewTaskListKey,
-          spaceId: spaceId,
-          canString: 'CanPostTaskList',
-          onPressed: () => showCreateUpdateTaskListBottomSheet(
-            context,
-            initialSelectedSpace: spaceId,
+        if (widget.onSelectTaskListItem == null)
+          ValueListenableBuilder(
+            valueListenable: showCompletedTask,
+            builder: (context, value, child) {
+              return TextButton.icon(
+                onPressed: () => showCompletedTask.value = !value,
+                icon: Icon(
+                  value
+                      ? Icons.visibility_off_outlined
+                      : Icons.visibility_outlined,
+                  size: 18,
+                ),
+                label: Text(value ? lang.hideCompleted : lang.showCompleted),
+                style: OutlinedButton.styleFrom(
+                  padding: const EdgeInsets.symmetric(horizontal: 10),
+                ),
+              );
+            },
+          ),
+        if (widget.onSelectTaskListItem == null)
```

Comment:
can be more legible by spread operator
```suggestion
        if (widget.onSelectTaskListItem == null) ...[
  ValueListenableBuilder(
    valueListenable: showCompletedTask,
    builder: (context, value, child) => TextButton.icon(
      onPressed: () => showCompletedTask.value = !value,
      icon: Icon(
        value ? Icons.visibility_off_outlined : Icons.visibility_outlined,
        size: 18,
      ),
      label: Text(value ? lang.hideCompleted : lang.showCompleted),
      style: OutlinedButton.styleFrom(
        padding: const EdgeInsets.symmetric(horizontal: 10),
      ),
    ),
  ),
  AddButtonWithCanPermission(
    key: TasksListPage.createNewTaskListKey,
    spaceId: spaceId,
    canString: 'CanPostTaskList',
    onPressed: () => showCreateUpdateTaskListBottomSheet(
      context,
      initialSelectedSpace: spaceId,
    ),
  ),
],
```